<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Qian Feng" />

<meta name="date" content="2018-01-04" />

<title>Phylogenetic network</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}

.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paper-reading</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/qianfeng2/workflowr">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Phylogenetic network</h1>
<h4 class="author"><em>Qian Feng</em></h4>
<h4 class="date"><em>2018-01-04</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-01-30</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 21669fb</p>
<!-- Add your analysis here -->
<section id="basic-concepts" class="level2">
<h2>Basic concepts</h2>
<p>Reassortment is the mixing of the genetic material of a species into new combinations in different individuals. It is particularly used when two similar viruses that are infecting the same cell exchange genetic material.</p>
<p>There are several interesting concepts in graph theory. Please look at the Figure 1.1, 1.2 and 1.3.</p>
<ul>
<li><p>The <strong>degree</strong>of node <span class="math inline">\(u\)</span> is the sum of its indegree and outdegree.</p></li>
<li><p>A <strong>biconnected components</strong> is the maximal subgraph that is induced by set of edges and doesn’t contain a cut node. A good example is in Figure 1.3.</p></li>
<li><p>A graph <span class="math inline">\(G=(V,E)\)</span> is called <strong>bipartite</strong> if and only if its set of nodes can be partitioned into subsets <span class="math inline">\(V_{1}\)</span>, <span class="math inline">\(V_{2}\)</span>,with <span class="math inline">\(V=V_{1}\cup V_{2}\)</span> and <span class="math inline">\(V_{1}\cap V_{2}=\emptyset\)</span>, such that for every edge <span class="math inline">\(e \in E\)</span>,one of the endpoints lies in <span class="math inline">\(V_{1}\)</span>, another endpoint lies in <span class="math inline">\(V_{2}\)</span>. Explain Exercise 1.2.3.</p></li>
<li><p>Two nodes <span class="math inline">\(v\)</span> and <span class="math inline">\(w\)</span> are <strong>imcomparable</strong>, if neither node is lower than other; similarily, two edges <span class="math inline">\(e\)</span> and <span class="math inline">\(f\)</span> are imcomparable if neither is lower than other;</p></li>
<li><p>Different <strong>traversals</strong> give rise to different orders in which nodes are examined. Pay attention to preorder, postorder and breadth-first traversal in Figure 1.7. In particular, breadth-first traversal, please see reference <a href="https://baike.baidu.com/item/??/9796023?fr=aladdin">here</a>.</p></li>
</ul>
<p>PS: Nodes are also called vertices, edges are also called branches or arcs. Bifurcating Tree = Resolve Tree = Binary Tree</p>
<p>Let <span class="math inline">\(\chi=\left\{x_{1},x_{2},...x_{n} \right\}\)</span> be a set of taxa, a <strong>cluster</strong> is any subset of <span class="math inline">\(\chi\)</span>,excluding the empty set <span class="math inline">\(\emptyset\)</span> and full set <span class="math inline">\(\chi\)</span>. The ultimate goal of phylogenetic analysis is to compute a set of clusters on <span class="math inline">\(\chi\)</span> such that each cluster is monophyletic(also called clade. Monophyletic group contains all descendants of the common ancestor and the ancestor itself).</p>
<p>A <strong>split</strong> is any bipartitioning of <span class="math inline">\(\chi\)</span> into two non-empty subsets <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span> of <span class="math inline">\(\chi\)</span>, such that <span class="math inline">\(\chi=A \cup B\)</span> and <span class="math inline">\(A\cap B=\emptyset\)</span>.</p>
<p>In phylogenetic analysis, a set of taxa <span class="math inline">\(\chi=\left\{x_{1},x_{2},...x_{n} \right\}\)</span> is often represented by a set of molecular sequences <span class="math inline">\(A=\left\{a_{1},a_{2},...a_{n} \right\}\)</span> where <span class="math inline">\(a_{i}\)</span> comes from taxon <span class="math inline">\(x_{i}\)</span> and correspond to some specific genes or locus. We also need to ensure that the sequences are <strong><em>homologous</em></strong>, that is, have evolved from a common ancestor sequence.</p>
<p>In <strong>Pairwise sequence alignment</strong>, with the help of substitution matrix, for example the BLOSSUM matrix, which assigns empirically score, we could calculate the score of each pair of residues and then sum over scores among all pairs would be the score of whole alignment.</p>
<p>Sequence are often aligned by inserting gaps into each sequence shown in Figure 2.6 such that all sequences have same length <span class="math inline">\(m\)</span>, forming a <strong>multiple sequence alignment </strong> of length <span class="math inline">\(m\)</span>. Our goal is to find a multiple sequence alignment that achieves the optimal score according to an appropriate score scheme. <em>Progressive method</em> as a heuristic approach, is used to align multiple sequences, its outline is shown in Figure 2.7. The core is to align a pair of similar sequences into <em>profiles</em>, then align profiles into final multiple sequence alignment.</p>
<p>Let <span class="math inline">\(M\)</span> be a multiple sequence alignment on <span class="math inline">\(\chi\)</span>, each column of <span class="math inline">\(M\)</span> is called a character, each symbol that occurs in this column is called a character state.</p>
</section>
<section id="phylogenetic-trees" class="level2">
<h2>Phylogenetic trees</h2>
<p>Now we are introducing some basic concepts and main methods for inferring phylogenetic trees.</p>
<p>Phylogenetic trees are usually computed from molecular sequences. They not only could uncover the relationship between different species or taxa, but also have many other applications. For instance, they are used to determine the age and the rate of diversification. In sequence-analysis method, they are allowed <strong>phylogenetic footprinting</strong>.</p>
<p>In practice, there are two types of analysis after the initial multiple sequence alignment: distance-based analysis and sequence-based one. Its outline is shown in Figure 3.1 at page 24.</p>
<section id="definition-phylogenetic-tree" class="level3">
<h3>Definition Phylogenetic Tree</h3>
<p>Given a set of taxa <span class="math inline">\(\chi\)</span>, this is a phylogenetic tree <span class="math inline">\(T=(V,E)\)</span> , its all nodes have degree <span class="math inline">\(\neq\)</span> 2, together with a taxon labeling <span class="math inline">\(\lambda:\chi\to V\)</span>that assigns actually one taxon to every leave and none to internal nodes.</p>
<p>From a theoretical and algorithmic point of view, unrooted phylogenetic trees are much more easier than rooted ones, however, in biology, rooted phylogenetic trees are usually more of interest. A phylogenetic tree is called an edge-weighted tree if we are given a map <span class="math inline">\(\omega\)</span> that assigns a non-negative weight or length <span class="math inline">\(\omega(e)\)</span> to every edge e of the tree. In drawings, we usually use length of the edge to indicate the scale rather than write the lengths explicitly next to edges.</p>
<p>Jukes-Cantor model tells us the probability formula of change during time <span class="math inline">\(t\)</span> or along the edge, given the mutation rate. This model of DNA evolution assumes the fours bases (A,C,G and T) occur with equal frequencies(0.25) and change from one base to another occurs at the same rate. If we relax the conditions, for example, let the bases occur at different and arbitrary rates (although they have to sum to 1), change rates in transitions and transversions, then we could get more generate model, anyway, they are both special cases of general time reversible model.</p>
<p>Classical phylogenetic trees construction approaches consist of two following types:</p>
<ul>
<li><strong>Sequence-based method</strong> usually searches for best phylogenetic tree which can optimally explain the given multiple sequence alignment <span class="math inline">\(M\)</span>. We discuss the three main approaches about it: maximum parsimony, ML and Bayesian inference.</li>
<li><strong>Distance-based method</strong> usually constructs phylogenetic tree from a given a distance matrix <span class="math inline">\(D\)</span>.</li>
</ul>
</section>
<section id="maximum-parsimony" class="level3">
<h3>Maximum parsimony</h3>
<p>Maximum parsimony method is to look for a phylogenetic tree that explains the given set of aligned sequences using a minimum number of evolutionary events.</p>
<p>The <em>parsimony score</em> of <span class="math inline">\(T\)</span>(tree) and <span class="math inline">\(M\)</span>(given multiple sequence alignment) is defined as:</p>
<span class="math display">\[\begin{equation}
P S(T, M)=\min_{\alpha}\sum_{\left\{x,y\right\}}diff(x,y)
\end{equation}\]</span>
<p>where diff() function is known as <em>hamming distance</em> between sequence <span class="math inline">\(x=(x_{1},x_{2},...x_{m})\)</span>and sequence <span class="math inline">\(y=(y_{1},y_{2},...y_{m})\)</span> that describes the difference of <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span> <span class="math display">\[\begin{equation}
diff(x, y)=| \left\{ i | x_{i}\neq y_{i} \right\} |
\end{equation}\]</span></p>
<p>The minimum is taken over all possible assignments <span class="math inline">\(\alpha\)</span> that make the sequences of length <span class="math inline">\(m\)</span> to be the internal nodes, summation is taken over all possible pairs of <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span> that are assigned at opposite end of edge of <span class="math inline">\(T\)</span> .</p>
<p>This task of computing parsimony score is the known <em>small parsimony</em> problem. For bifurcating tree, <em>Fitch algorithm</em> is used to calculate efficiently, in more general settings, <em>Sankoff’s algorithm</em> can be applied. Let’s describe Fitch algorithm. Assume we are given a multiple sequence alignment <span class="math inline">\(M\)</span> and a bifurcating tree <span class="math inline">\(T\)</span> on <span class="math inline">\(\chi\)</span>, we need to score each character (that is, column of the alignment) separately, and then obtain the parsimony score <span class="math inline">\(P S(T, M\)</span> by summing over all characters.</p>
</section>
<section id="bayesian-methods" class="level3">
<h3>Bayesian methods</h3>
<p>Bayesian inference employs MCMC to sample from the posterior probability distribution. Posterior probability distribution in phylogeny inference is the conditional probability of T given input dataset. Each step, Metropolis-Hastings algorithm is used here, each step we propose a modified new tree topology, then accept it with the probability of <span class="math inline">\(min(1, \alpha)\)</span>, <span class="math inline">\(\alpha\)</span> is the ratio of new prior times transition kernel divide the old one, otherwise, we reject and still use the current one. Therefore, a suitable modification of this proposed new tree should be taken into account with care. After discarding the burn in, sometimes we would also do sparsely sampling from output a series of trees, retaining only 1000th tree, say, in order to avoid the problem of autocorrelation, since the samples the not independent in this case. Let’s talk about the modification in detail.</p>
<p>The first one is called <em>local algorithm</em>, which is a modification of NNI(Nearest neighbor interchange). This algorithm is very magic and powerful. Through a path in a tree, then update length of different nodes and even topology by introducing random number. Regarding parameters in evolutionary model, like mutation rate <span class="math inline">\(\mu\)</span> in Jukes Cantor model, are modified by adding a random number uniformly chosen from an appropriate interval centered at 0. For a set of parameters that is constrained to sum to some specific value, such as 1 in the case of probabilities, the values are randomly modified according to a Dirichlet distribution.</p>
<p>Modifying the tree and proposal for new parameters are generated independently and simultaneously, and then accept or reject by a single Metropolis-Hastings ratio.</p>
<p>The second one is called <em>Metropolis-coupled MCMC</em> or (<span class="math inline">\(MC^3\)</span>) , this is a variant of MCMC which are suitable if there are a large amount of parameters. The first chain <span class="math inline">\(Z_{1}\)</span> is called cold chain, other chains <span class="math inline">\(Z_{2},...Z_{K}\)</span> are called heated chains, all chains are run parallel. After all chains have moved one step, Let <span class="math inline">\(T_{i}\)</span> denote the current state of all chains, for all i =1 ,2 … K, then after randomly choosing two chains <span class="math inline">\(Z_{i}\)</span> and <span class="math inline">\(Z_{j}\)</span>, swap the result with specific probability. Swapping of chains can help the cold chains to move a new part of of parameter space that it may have difficulty in reaching. At the end of run, discard all heated chains and cold chain is processed as before.</p>
<p>A challenge here is we may not know whether this chain has converged. Two approaches are introduced here: one is to monitor whether there there is no further increase in likelihood, another one is run multiple chains in parallel, to keep them running until appear to sample from the same distribution. In addition, areas like choosing prior distribution and check convergence is ongoing research.</p>
</section>
</section>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
      "HTML-CSS": { availableFonts: ["TeX"] }
    });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
