<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}

.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paper-reading</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xiangzhu/paper-reading">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>This is a website for organizing my reading notes of research papers.</p>
<p>If you have any question about this project, please feel free to contact me: <a href="https://github.com/xiangzhu">Xiang Zhu</a>, <code>xiangzhu@stanford.edu</code>.</p>
<hr />
<section id="integrative-genomics" class="level2">
<h2>Integrative Genomics</h2>
<section id="peca-model" class="level3">
<h3><a href="">PECA model</a></h3>
<ul>
<li>Duren, Z., Chen, X., Jiang, R., Wang, Y., &amp; Wong, W. H. (2017). Modeling gene regulation from paired expression and chromatin accessibility data. <em>Proc. Natl. Acad. Sci. U.S.A.</em></li>
<li>Paper: <a href="http://www.pnas.org/content/114/25/E4914" class="uri">http://www.pnas.org/content/114/25/E4914</a></li>
<li>Software: <a href="http://web.stanford.edu/~zduren/PECA/" class="uri">http://web.stanford.edu/~zduren/PECA/</a></li>
</ul>
</section>
<section id="omnigenic-model" class="level3">
<h3><a href="">Omnigenic model</a></h3>
<ul>
<li>Boyle, E. A., Li, Y. I., &amp; Pritchard, J. K. (2017). An Expanded View of Complex Traits: From Polygenic to Omnigenic. <em>Cell</em>.</li>
<li>Paper: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28622505" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/28622505</a></li>
</ul>
<hr />
</section>
</section>
<section id="electronic-medicalhealth-records" class="level2">
<h2>Electronic Medical/Health Records</h2>
<section id="analysis-of-insurance-claims" class="level3">
<h3><a href="">Analysis of insurance claims</a></h3>
<ul>
<li>Wang, K., Gaitsch, H., Poon, H., Cox, N. J., &amp; Rzhetsky, A. (2017). Classification of common human diseases derived from shared genetic and environmental determinants. <em>Nat. Genet.</em></li>
<li>Paper: <a href="http://www.nature.com/ng/journal/v49/n9/full/ng.3931.html" class="uri">http://www.nature.com/ng/journal/v49/n9/full/ng.3931.html</a></li>
</ul>
</section>
<section id="treewas-model" class="level3">
<h3><a href="">TreeWAS model</a></h3>
<ul>
<li>Cortes, A., Dendrou, C., Motyer, A., Jostins, L., Vukcevic, D., Dilthey, A., Donnelly, P., Leslie, S., Fugger, L., &amp; McVean, G. (2017). Bayesian analysis of genetic association across tree-structured routine healthcare data in the UK Biobank. <em>Nat. Genet.</em></li>
<li>Paper: <a href="https://www.nature.com/ng/journal/v49/n9/full/ng.3926.html" class="uri">https://www.nature.com/ng/journal/v49/n9/full/ng.3926.html</a></li>
<li>Software: <a href="https://github.com/mcveanlab/TreeWAS" class="uri">https://github.com/mcveanlab/TreeWAS</a></li>
</ul>
</section>
<section id="stochastic-compartmental-model-for-infection" class="level3">
<h3><a href="">Stochastic compartmental model for infection</a></h3>
<ul>
<li>Cule, M. &amp; Donnelly, P. (2017). Stochastic modelling and inference in electronic hospital databases for the spread of infections: Clostridium difficile transmission in Oxfordshire hospitals 2007–2010. <em>Ann. Appl. Stat.</em></li>
<li>Paper: <a href="https://projecteuclid.org/euclid.aoas/1500537718" class="uri">https://projecteuclid.org/euclid.aoas/1500537718</a></li>
</ul>
<hr />
</section>
</section>
<section id="han-chinese-population-genetics" class="level2">
<h2>Han Chinese Population Genetics</h2>
<section id="major-depressive-disorder-gwas" class="level3">
<h3>Major depressive disorder GWAS</h3>
<ul>
<li>Converge Consortium. (2015). Sparse whole genome sequencing identifies two loci for major depressive disorder. <em>Nature.</em></li>
<li>Paper: <a href="http://www.nature.com/nature/journal/v523/n7562/full/nature14659.html" class="uri">http://www.nature.com/nature/journal/v523/n7562/full/nature14659.html</a></li>
</ul>
</section>
<section id="genetic-variation" class="level3">
<h3>Genetic variation</h3>
<ul>
<li>Charleston W. K. Chiang, Serghei Mangul, Christopher R. Robles, Warren W. Kretzschmar, Na Cai, Kenneth S. Kendler, Sriram Sankararam, Jonathan Flint. (2017). A comprehensive map of genetic variation in the world’s largest ethnic group - Han Chinese. <em>bioRxiv.</em></li>
<li>Preprint: <a href="https://www.biorxiv.org/content/early/2017/07/13/162982" class="uri">https://www.biorxiv.org/content/early/2017/07/13/162982</a></li>
</ul>
<hr />
</section>
</section>
<section id="statistical-genetics" class="level2">
<h2>Statistical Genetics</h2>
<section id="an-incomplete-list-of-fine-mapping-methods" class="level3">
<h3><a href="fine_mapping">An incomplete list of fine mapping methods</a></h3>
</section>
</section>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
      "HTML-CSS": { availableFonts: ["TeX"] }
    });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
