<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}

.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paper-reading</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/qianfeng2/workflowr">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>This is a website for organizing my reading notes of research papers and a few books.</p>
<p>If you have any question about this project, please feel free to contact me: <a href="https://qianfeng2.github.io">Qian Feng</a>, <code>fengq2@student.unimelb.edu.au</code>.</p>
<hr />
<div id="paper-review" class="section level2">
<h2>Paper review</h2>
<div id="general-review-papers" class="section level3">
<h3>general review papers</h3>
<div id="huson-2011" class="section level4">
<h4><a href="">Huson 2011</a></h4>
<ul>
<li>A survey of combinatorial methods for phylogenetic networks. <em>Huson and Scornavacca, Genome Biol. Evol.2011</em>.</li>
<li>Paper: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3017387/pdf/evq077.pdf" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3017387/pdf/evq077.pdf</a></li>
<li>Note: <a href="Huson2011paper.html">link</a></li>
</ul>
</div>
<div id="david-a.morrison-2014" class="section level4">
<h4><a href="">David A.Morrison 2014</a></h4>
<ul>
<li>Phylogenetic networks: a review of methods to display evolutionary history <em>David A.Morrison, Annual Research and Review in Biology, 2014</em></li>
<li>Paper: <a href="http://www.journalrepository.org/media/journals/ARRB_32/2014/Jan/Morrison4102013ARRB8230_1.pdf" class="uri">http://www.journalrepository.org/media/journals/ARRB_32/2014/Jan/Morrison4102013ARRB8230_1.pdf</a></li>
<li>Note: <a href="morrison2014paper.html">link</a></li>
</ul>
</div>
</div>
<div id="identify-recombination" class="section level3">
<h3>Identify recombination</h3>
<div id="average-pac-likelihood-using-snps" class="section level4">
<h4><a href="">Average PAC likelihood using SNPs</a></h4>
<ul>
<li>Modeling linkage disequilibrium and identifying recombination hotspots using single-nucleotide polymorphism data. <em>Genetics 165.4 (2003): 2213-2233.Na Li and Matthew Stephens.</em>.</li>
<li>Paper: <a href="http://www.genetics.org/content/genetics/165/4/2213.full.pdf" class="uri">http://www.genetics.org/content/genetics/165/4/2213.full.pdf</a></li>
<li>Note: <a href="PAC.html">link</a></li>
</ul>
</div>
<div id="integrative-software-rdp" class="section level4">
<h4><a href="">Integrative Software RDP</a></h4>
<ul>
<li>RDP4: Detection and analysis of recombination patterns in virus genomes <em>Virus Evolution, 1(1),(2015)Martin, D. P., Murrell, B., Golden, M., Khoosal, A., &amp; Muhire, B.</em></li>
<li>Paper: <a href="https://academic.oup.com/ve/article/1/1/vev003/2568683" class="uri">https://academic.oup.com/ve/article/1/1/vev003/2568683</a></li>
<li>Software: <a href="http://web.cbio.uct.ac.za/~darren/RDP4Manual.pdf" class="uri">http://web.cbio.uct.ac.za/~darren/RDP4Manual.pdf</a></li>
<li>Note: <a href="RDP4.html">link</a></li>
</ul>
</div>
<div id="lower-and-upper-bound-on-minimum-recombination-number" class="section level4">
<h4><a href="">Lower and upper bound on minimum recombination number</a></h4>
<ul>
<li>Efficient computation of close lower and upper bounds on the minimum number of recombinations in biological sequence evolution. <em>Song, Yun S., Yufeng Wu, and Dan Gusfield. Bioinformatics 21.suppl_1 (2005): i413-i422.</em></li>
<li>Paper: <a href="https://academic.oup.com/bioinformatics/article/21/suppl_1/i413/203304" class="uri">https://academic.oup.com/bioinformatics/article/21/suppl_1/i413/203304</a></li>
<li>Note: <a href="">link</a></li>
</ul>
</div>
<div id="recombination-estmation-using-hmm-and-pac" class="section level4">
<h4><a href="">Recombination estmation using HMM and PAC</a></h4>
<ul>
<li>Hypervariable antigen genes in malaria have ancient roots. <em>Zilversmit, Martine M., et al. BMC evolutionary biology 13.1 (2013): 110.</em>.</li>
<li>Paper: <a href="https://bmcevolbiol.biomedcentral.com/articles/10.1186/1471-2148-13-110" class="uri">https://bmcevolbiol.biomedcentral.com/articles/10.1186/1471-2148-13-110</a></li>
<li>Note: <a href="zilversmitpaper.html">link</a></li>
<li>Practice:<a href="">link</a></li>
</ul>
</div>
</div>
<div id="phylogenetic-network-construction" class="section level3">
<h3>Phylogenetic Network construction</h3>
<div id="algorithm-from-a-given-mul-tree" class="section level4">
<h4><a href="">Algorithm from a given MUL tree</a></h4>
<ul>
<li>Reconstructing the evolutionary history of polyploid from multilabeled trees. <em>Huber, Katharina T, et al. Molecular Biology and Evolution 23.9 (2006): 1784-1791.</em></li>
<li>Paper: <a href="https://academic.oup.com/mbe/article/23/9/1784/1014292" class="uri">https://academic.oup.com/mbe/article/23/9/1784/1014292</a></li>
<li>Note: <a href="MUL.html">link</a></li>
</ul>
</div>
<div id="joint-baysian-estimation-of-alignment-and-phylogeny" class="section level4">
<h4><a href="">Joint Baysian Estimation of Alignment and Phylogeny</a></h4>
<ul>
<li>1.Joint Bayesian estimation of alignment and phylogeny. <em>Redelings, Benjamin D., and Marc A. Suchard. Systematic biology 54.3 (2005): 401-418.</em></li>
<li>Paper: <a href="https://academic.oup.com/sysbio/article/54/3/401/1727336" class="uri">https://academic.oup.com/sysbio/article/54/3/401/1727336</a></li>
<li><p>Note: <a href="jointestimation.html">link</a></p></li>
<li>2.BayesCAT: Bayesian co‐estimation of alignment and tree. <em>Shim, Heejung, and Bret Larget. Biometrics (2017).</em></li>
<li>Paper: <a href="https://heejungshim.org/research/#BayesCAT" class="uri">https://heejungshim.org/research/#BayesCAT</a></li>
<li><p>Note: <a href="">link</a></p></li>
</ul>
</div>
</div>
<div id="var-biology-background-knowledge" class="section level3">
<h3>Var biology background knowledge</h3>
<ul>
<li>1.Ape parasite origins of human malaria virulence genes. <em>Larremore, D. B. et al. Nat. Commun. 6:8368 doi: 10.1038/ncomms9368 (2015)</em></li>
<li>Paper: <a href="https://www.nature.com/articles/ncomms9368.pdf" class="uri">https://www.nature.com/articles/ncomms9368.pdf</a></li>
<li><p>Note: <a href="larremore2016.html">link</a></p></li>
<li>2.Plasmodium falciparum Erythrocyte Membrane Protein 1 Diversity in Seven Genomes – Divide and Conquer. <em>Rask, Thomas S., et al.PLoS computational biology 6.9 (2010): e1000933.</em></li>
<li>Paper: <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000933" class="uri">http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000933</a></li>
<li><p>Note: <a href="Raskpaper.html">link</a></p></li>
</ul>
<hr />
</div>
</div>
<div id="magic-algorithms" class="section level2">
<h2>Magic Algorithms</h2>
<div id="mcmc" class="section level4">
<h4><a href="">MCMC</a></h4>
<ul>
<li>Understanding the metropolis-hastings algorithm. <em>Chib, Siddhartha, and Edward Greenberg.The american statistician 49.4 (1995): 327-335.</em></li>
<li>Paper: <a href="http://www2.stat.duke.edu/~scs/Courses/Stat376/Papers/Basic/ChibGreenberg1995.pdf" class="uri">http://www2.stat.duke.edu/~scs/Courses/Stat376/Papers/Basic/ChibGreenberg1995.pdf</a></li>
<li>Note: <a href="MCMC.html">link</a></li>
<li>Random walk example: <a href="randomwalk.html">link</a></li>
<li>Simulated annealing: <a href="https://github.com/qianfeng2/PNcode/tree/master/code/simulated%20annealing">link</a></li>
<li>A short and useful video:<a href="https://www.youtube.com/watch?v=a_08GKWHFWo">Gibbs sampling</a></li>
</ul>
<hr />
</div>
</div>
<div id="hmm-and-phmm-model" class="section level2">
<h2>HMM and PHMM model</h2>
<div id="drive-hmm-and-phmm-in-r" class="section level4">
<h4><a href="PHMM.html">Drive HMM and PHMM in R</a></h4>
<hr />
</div>
</div>
<div id="book-review" class="section level2">
<h2>Book Review</h2>
<div class="figure">
<img src="cover.png" alt="Book Cover" />
<p class="caption">Book Cover</p>
</div>
<div id="book1-phylogenetic-networks-concepts-algorithms-and-applications" class="section level3">
<h3><a href="book1.html">Book1: Phylogenetic networks: concepts, algorithms and applications</a></h3>
<div class="figure">
<img src="book_cover.jpg" alt="Book Cover" />
<p class="caption">Book Cover</p>
</div>
</div>
<div id="book2-gene-genealogies-variation-and-evolution" class="section level3">
<h3><a href="https://qianfeng2.github.io/Others/coalescent_booksummary.pdf">Book2: Gene Genealogies, Variation and Evolution</a></h3>
<div class="figure">
<img src="ML.jpg" alt="Book Cover" />
<p class="caption">Book Cover</p>
</div>
</div>
<div id="book3-machine-learning.-a-probabilistic-perspective" class="section level3">
<h3><a href="ML.html">Book3: Machine Learning. A Probabilistic Perspective</a></h3>
<hr />
</div>
</div>
<div id="plot-tools" class="section level2">
<h2>Plot tools</h2>
<div id="play-with-plotly-r-package" class="section level3">
<h3><a href="plotly.html">Play with plotly R package</a></h3>
<hr />
</div>
</div>
<div id="computer-science" class="section level2">
<h2>Computer science</h2>
<div id="algorithm-complexity-big-o" class="section level3">
<h3><a href="https://rob-bell.net/2009/06/a-beginners-guide-to-big-o-notation">Algorithm complexity: Big O</a></h3>
</div>
</div>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
      "HTML-CSS": { availableFonts: ["TeX"] }
    });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
