<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Qian Feng" />

<meta name="date" content="2018-01-16" />

<title>Joint bayesian estimation of alignment and phylogeny</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}

.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paper-reading</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/qianfeng2/workflowr">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Joint bayesian estimation of alignment and phylogeny</h1>
<h4 class="author"><em>Qian Feng</em></h4>
<h4 class="date"><em>2018-01-16</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-01-25</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 3a6eca1</p>
<!-- Add your analysis here -->
<p>In this paper, authors propose a novel model to estimate multiple sequence alignment , phylogenetic tree reconstruction and their support simultaneously. They operate in Bayesian framework and use MCMC methods.</p>
<p>In general, researchers first need to finish multiple sequence alignment, and then use it to reconstruct the phylogeny. However, if the alignment contains ambiguous regions particular like in distantly related sequences, this would lead to inaccurate result. A normal technique is to remove these regions, hence, leading to a loss of a large fraction of informative sites. Due to this, researchers come up with a simple method: split ambiguous columns into groups of residues in which homology is unambiguous, then, place them in separate columns, yet, it is still worth noting that identifying these ambiguous regions is too subjective.</p>
<p>There are a number of techniques are developed to get a better use of ambiguous alignment. One technique, known as elision, concatenates a set of near-optimal alignments into a larger alignment and use them for reconstructing phylogeny. However, elision treat all the near optimal alignment equally instead of weighted, as a consequence, it may be not so good. Another technique, known as optimization alignment, involves estimation of alignment and phylogenies simultaneously with parsimony framework. One problem of optimization alignment is the measure of uncertainty is hard to obtain since standard bootstrap couldn’t be applied due to the dependent alignment columns. Method in this paper not only weighted the alignment naturally, but also assess the confidence using posterior probability.</p>
<p>Advantages of joint estimation are as follows: one thing is joint estimation doesn’t require extra guide tree, this contrast with alignment with progressive alignment are biased and need a guide tree. Another thing is joint estimation could provide more accurate substitution and indel (inseration/deletion) models, in scoring alignment by extended substitution model, in using shared indels to group taxa on a tree.</p>
<p>In developing indel model, there is a simplified assumption: indel event occurs independent on each branch. The reason of this is it allows us to use a simple pair-HMM to model the alignment. In addition, insertions and deletions are equally likely and sequence lengths do not grow or shrink over time. In substitution model, evolution is independent across columns of <span class="math inline">\(f\)</span> and each branch.</p>
<p>MCMC needs to sample from posterior distribution of the alignment, phylogeny and model parameters given only unaligned sequences. In this paper, researchers introduce a new transition kernel to resample the topology and alignment that improves mixing efficiency, allowing chains to converge even when start with an arbitrary alignment.</p>
<p>Multiple sequence alignment <span class="math inline">\(A\)</span> is actually a matrix <span class="math inline">\(f\)</span>, it specifies which letters from sequences are homologous by arranging homologous letters into the same column in this matrix.</p>
<ul>
<li>Table: This is an annotation list</li>
</ul>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Item</th>
<th style="text-align: center;">Name</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: center;"><span class="math inline">\(Y\)</span></td>
<td>a set of <span class="math inline">\(n\)</span> homologous molecular sequences</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: center;"><span class="math inline">\(A\)</span></td>
<td>multiple sequence alignment</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: center;"><span class="math inline">\(\tau\)</span></td>
<td>unrooted tree topology</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: center;"><span class="math inline">\(T\)</span></td>
<td>branch length</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: center;"><span class="math inline">\(\Theta\)</span></td>
<td>substitution process parameters</td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td style="text-align: center;"><span class="math inline">\(\Lambda\)</span></td>
<td>indel process parameters</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: center;"><span class="math inline">\(N\)</span></td>
<td>total number of nodes in <span class="math inline">\(\tau\)</span>: <span class="math inline">\(N=2n-2\)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: center;"><span class="math inline">\(B\)</span></td>
<td>total number of branches in <span class="math inline">\(\tau\)</span>: <span class="math inline">\(B=2n-3\)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">9</td>
<td style="text-align: center;"><span class="math inline">\(\gamma\)</span></td>
<td>distribution of ancestral letters at the root node</td>
</tr>
<tr class="even">
<td style="text-align: left;">10</td>
<td style="text-align: center;"><span class="math inline">\(b\)</span></td>
<td>every branch</td>
</tr>
<tr class="odd">
<td style="text-align: left;">11</td>
<td style="text-align: center;"><span class="math inline">\(\rho(b)\)</span></td>
<td>parent branch for b</td>
</tr>
<tr class="even">
<td style="text-align: left;">12</td>
<td style="text-align: center;"><span class="math inline">\(n(b)\)</span></td>
<td>node in <span class="math inline">\(\tau\)</span> shared by the branch b and <span class="math inline">\(\rho(b)\)</span></td>
</tr>
</tbody>
</table>
<p>Therefore, the whole state space <span class="math inline">\(\Omega\)</span> is composed of points: <span class="math inline">\(\omega=(Y, A, \tau, T,\Theta,\Lambda )\)</span>.</p>
<section id="substitution-model" class="level2">
<h2>Substitution model</h2>
<p>Likelihood <span class="math inline">\(P(Y | A,\tau,T,\Theta,\Lambda)\)</span> is given in substitution model, before getting this result, we need to accomplish two tasks, first one is to specify how <span class="math inline">\(A\)</span> arranges <span class="math inline">\(Y\)</span> into matrix <span class="math inline">\(f\)</span>, next one is to specify the probabilistic model on the columns of <span class="math inline">\(f\)</span>. Figure 1 in the original paper solves above first task, the tuples at the leaf node within a column in matrix <span class="math inline">\(f\)</span> are from a multinomial distribution which addresses second task mentioned before. An worth noting thing is there is <strong>Felsenstein wildcards</strong> in matrix <span class="math inline">\(f\)</span> which represents internal nodes that are present but unobserved denoted by <span class="math inline">\(*\)</span>. The full likelihood is by multiplying likelihood of each column using peeling algorithm, meanwhile Felsenstein wildcards and gaps are treated as missing data.</p>
<p>A Markov model is reversible equivalent to hold the detail balance equation: <span class="math display">\[\begin{equation}
\pi_{i}p_{ij}=\pi_{j}p_{ji}
\end{equation}\]</span></p>
</section>
<section id="gap-model" class="level2">
<h2>Gap model</h2>
<p>We describe prior of indel process parameters <span class="math inline">\(\Lambda\)</span> in gap model. In gap model, <span class="math inline">\(A\)</span> could be represented as a tuple pairwise alignments: <span class="math inline">\((A^{(1)}, A^{(2)},..., A^{(B)})\)</span>, we replace the alignment prior (equation(11)) with standard prior (equation(10)), then after calculating this modified posterior, Gibbs sample from it using DP programs, detailed DP algorithm is provided in Appendix.</p>
<p>Three parameters are used in pair-HMM, including <span class="math inline">\(\delta\)</span>,<span class="math inline">\(\epsilon\)</span>and <span class="math inline">\(\zeta\)</span>,parameter <span class="math inline">\(\delta\)</span> refers to the probability of an indel in either sequence, we assume double-exponential distribution on the approximate log odds of <span class="math inline">\(\delta\)</span>; <span class="math inline">\(\epsilon\)</span> refers to the probability of extending an existing gap, and exponential distribution is assumed, for <span class="math inline">\(\zeta\)</span>, it means the transition probability from any state to the end state, in our example, <span class="math inline">\(\zeta=1/1000\)</span>.</p>
</section>
<section id="mcmc-sampling-process" class="level2">
<h2>MCMC sampling process</h2>
<p>In MCMC sampling process, researchers employ a <strong>random scan Metropolis-within-Gibbs</strong> approach. In every iteration, they attempt to sample from every model parameters at least once. Indel parameters are resampled more frequently than substitution parameters. NNI is used to update topology <span class="math inline">\(\tau\)</span>, it will move across every internal node at least once per iteration.</p>
<p>In whole, topology <span class="math inline">\(\tau\)</span> is updated by a number of MH steps, each alters only part of the topology. Once a topology is chosen, the internal nodes are resampled from the DP matrix, after this step, alignment <span class="math inline">\(A\)</span> and topology <span class="math inline">\(\tau\)</span> is resampled again. The MH acceptance probability is given in equation (12). This is an 1D DP problem.</p>
<p>Regarding to alignment sampling, traditional two MCMC transition kernels are not efficient enough in some circumstances shown in Figure 4, researchers decide to resample the alignment along a branch and the sequence at one end of this branch in the same step, this is a 2D DP problem.</p>
<p>Alignment uncertainty plot (AU) is introduced to depict the alignment variability, it is drawn from the posterior alignment, and provides a valuable tool to assess alignment ambiguity.</p>
<p>We show this modified MCMC algorithm converges to equilibrium distribution more quickly than previously available MCMC transition kernel. This allows us to choose randomly one alignment rather than start from an estimated alignment via other software like ClustalW, which is the first advantage, another benefit is that it would decrease burn-in time and less autocorrelation. In order to assess the convergence of continuous parameters, Gelman-Rubin R statistics is employed, results suggesting each chain converges to the same distribution, 95% Baysian credible intervals about posterior probability (PP) are also given in table 2.</p>
</section>
<section id="bali-phy-show" class="level2">
<h2>BAli-Phy show</h2>
<p>BAli-Phy is a C++ software providing samples from posterior of alignment and phylogeny model. I will show it step by step to interested readers. This software please click <a href="http://www.bali-phy.org/index.php">here</a>.</p>
<p>Firstly install the BAli-Phy using homebrew based on its <a href="http://www.bali-phy.org/README.html#running">User’s Guide</a> in this software website. We will run commands in terminal. Type this command to check its version: <code>bali-phy --version</code></p>
<p>Next step is to install programs used for viewing the results<br />
1. Tracer : MCMC parameter, diagnostic viewer.<br />
2. FigTree : Phylogeny Viewer<br />
3. SeaView : Alignment viewer.</p>
<p>Then a quick start to run BAli-Phy is to type in two following commands:<br />
<code>bali-phy ~/examples/sequences/5S-rRNA/5d.fasta --iter=150</code><br />
<code>bp-analyze 5d-1/</code></p>
<p>The output results are in a separate file named “5d-1”&quot; which is in root directory (~/fengqian), including</p>
<ul>
<li>Table: This is a result list</li>
</ul>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Item</th>
<th style="text-align: center;">Name</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: center;">C1.log</td>
<td>Numeric parameters: indel and substitution rates, etc. Opened by Tracer</td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td style="text-align: center;">C1.trees</td>
<td>Tree samples: one sample per line, in Newick format. Opened by FigTree</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td style="text-align: center;">C1.Pp.fastas</td>
<td>Sampled alignments for partition p including ancestral sequences. Opened by SeaView</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: center;">C1.out</td>
<td>Iteration numbers, probabilities, success probabilities for transition kernels.</td>
</tr>
</tbody>
</table>
<p>and other little files.</p>
<p>Last, to summarize the output, type this to find majority consensus tree, note the dir has changed to 5d-1.<br />
<code>trees-consensus C1.trees $&gt;$ c50.PP.tree</code></p>
<p>To compute the maximum a posteriori tree, input:<br />
<code>trees-consensus --skip=10% C1.trees --map-tree=MAP.tree</code></p>
<p>Checking topology convergence use: <code>trees-bootstrap C1.trees</code><br />
It will show us the PP and LOD values regarding to each topology as well.</p>
<p>Anyway, even though this framework enjoys great advantages in estimating alignment and phylogeny at the same time, there exist some limitations as well. First is the parameters prior are not sufficient for biological meaning. Another shortcoming is indel process parameters are the same along each branch, which contrasts with ClustalW. In ClustalW, there are a function of branch lengths.</p>
<p>It is a challenge for me to fully understand equation (4) and (6) in this paper.</p>
<p>PS: I am learning MCMC from Chib Siddhartha and Edward Greenberg (1995)’s <a href="http://www2.stat.duke.edu/~scs/Courses/Stat376/Papers/Basic/ChibGreenberg1995.pdf">paper</a>, a powerful algorithm, it seems pretty interesting.^_^</p>
</section>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
      "HTML-CSS": { availableFonts: ["TeX"] }
    });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
